tools/cert_create/Makefile

```diff
@@ -61,8 +54,7 @@ all: clean ${BINARY}
 
 ${BINARY}: ${OBJECTS} Makefile
 	@echo "  LD      $@"
-	@echo 'const char build_msg[] = "Built : "__TIME__", "__DATE__; \
-                const char platform_msg[] = "${PLAT}";' | \
```

Comment:
I refute the suggestion that TBBR is _designed_ to only work on expensive devices. As stated in https://github.com/ARM-software/tf-issues/issues/448, there are are number of things that can be done to improve the implementation but these are not fundamental design issues.

Regarding OID compatibility, it's very difficult to know how the codebase is being used, since most vendor's platform code is not upstream. It's not a simple case of vendors either using the upstream TBBR implementation as-is or not at all. Many vendors pick and choose the code they need, which may or may not result in a non-TBBR compliant implementation. But we still need to try to provide compatibility where possible.

For example, it's possible that some vendors are using the upstream device side code (or a forked version of it) but their own certificate creation tools. Or their device side code version may not be in lock-step with their cert_create version. Silently changing the OIDs under their feet could easily result in authentication failures. Doing this with no migration path is not acceptable (in my opinion). Also, some vendors may simply prefer to use their own OIDs, especially in the case where their various firmware images are owned by different entities.

I'm happy for non-ARM vendors to use the ARM Limited OIDs if they want, and I'm also happy for this to eventually be the default behaviour. But I don't want to force this on all vendors.

Regarding "how long should we carry USE_TBBR_DEFS?", I think some vendors will always need this option, although the default option can change from false to true at some point. That "some point" is likely to be when we make a v2.0 release, at which time we can remove all conditional code under the ERROR_DEPRECATED flag.